Added the endpoint to download a dataset resource

ahervy opened the pull request:
## Summary

This pull request adds the endpoint to download a dataset resource in the docs.

Story details: https://app.shortcut.com/opendatasoft/story/29796

## Changes 

- Added the docs for the endpoint in the Dataset Resources section.
- Added the required permissions.

Commit:
Added the endpoint to download a dataset resource

- Added the docs for the endpoint
- Added the required permissions

bost-h commented on source/includes/_dataset_resources.md:
```diff
@@ -209,6 +209,36 @@ The payload must be a valid resource object without any uid.
 
+## Download a resource
+
+This API endpoint is meant to download a resource specified with its UID.
```
> download a resource specified with its UID.

And below:

> Returns: The resource with the specified UID

To me, in the context of this documentation, a resource is the Opendatasoft object linked to a dataset and this endpoint is to download the file linked to this resource.

Personally I find it more clear to say "the file used by the resource" or something like that, instead of "the resource" , since this route doesn't return the "opendatasoft resource object linked to a dataset", but maybe it's just me

ahervy replied:
Got it, I'll change the wording a suggested (i.e. "the file used by the resource").

bost-h commented on source/includes/_permissions.md:
```diff
@@ -326,6 +326,15 @@ Requires one of the following:
+Requires one of the following:
+
+- domain permission: `edit_dataset`
+- current dataset permission: `edit_dataset` or `publish_dataset`
```
The permission to publish the dataset doesn't allow the user to download a resource file

ahervy replied:
OK, I'll remove the `publish_dataset` permission from this line.

bost-h reviewed (changes requested):
Thanks for the documentation 👍 

Just a blocking comment about the permission required to access this endpoint

Commit:
Updated the docs based on PR comments

- Removed the publish_dataset permission from the required permissions
- Changed the wording for the file (file used by the resource)

bost-h reviewed (approved):
👍
